.github/workflows/review-check.yml

```diff
@@ -10,47 +10,54 @@ jobs:
 
     steps:
     - name: Checkout code
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
 
-    - name: Get list of files changed
-      id: files
-      run: |
-        echo "::set-output name=files::$(git diff --name-only ${{ github.event.before }} ${{ github.sha }} | tr '\n' ' ')"
-    
     - name: Check if PR modifies specific path
+      shell: bash
       id: check-path
       run: |
-        if echo "${{ steps.files.outputs.files }}" | grep -q '^src/'; then
-          echo "::set-output name=specific_path::true"
+        BASE_SHA=$(jq -r .pull_request.base.sha < $GITHUB_EVENT_PATH)
+        HEAD_SHA=$(jq -r .pull_request.head.sha < $GITHUB_EVENT_PATH)
+        git fetch origin $BASE_SHA $HEAD_SHA
+        files=$(git diff --name-only $BASE_SHA $HEAD_SHA | tr '\n' ' ')
+        echo "files=$files" >> $GITHUB_OUTPUT
+        if echo "$files" | grep -q 'src/contracts/'; then
+          echo "specific_path=true" >> $GITHUB_OUTPUT
         else
-          echo "::set-output name=specific_path::false"
+          echo "specific_path=false" >> $GITHUB_OUTPUT
         fi
 
     - name: Ensure required reviewers
+      shell: bash
       id: ensure-reviewers
       run: |
         GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}
         PR_NUMBER=${{ github.event.pull_request.number }}
         REPO=${{ github.repository }}
         
-        # Fetch reviewers of the PR
+        # Fetch approved and non-dismissed reviews of the PR
         REVIEWERS=$(curl -s -H "Authorization: token $GITHUB_TOKEN" \
                     "https://api.github.com/repos/$REPO/pulls/$PR_NUMBER/reviews" \
-                    | jq -r '.[].user.login')
+                    | jq -r '.[] | select(.state == "APPROVED" and .dismissed_at == null) | .user.login')
```

Comment:
Just in case a review isn't dismissed when another one is submitted immediately afterwards.

```suggestion
                    | jq -r '[.[] | select(.state == "APPROVED" and .dismissed_at == null) | .user.login] | unique | .[]')
```

<details><summary>Test</summary>

(Note that I removed the check for "dismissed_at".)
```
curl -s "https://api.github.com/repos/cowdao-grants/cow-fee/pulls/21/reviews" | jq -r '[.[] | select(.state == "APPROVED") | .user.login] | unique | .[]'
fedgiac
fleupold
mfw78
```
</details>